ebean-core/src/main/java/io/ebeaninternal/server/querydefn/DefaultOrmQuery.java

```diff
@@ -616,7 +616,7 @@ public SpiQuerySecondary convertJoins() {
    * Limit the number of fetch joins to Many properties, mark as query joins as needed.
    */
   private void markQueryJoins() {
-    detail.markQueryJoins(beanDescriptor, lazyLoadManyPath, isAllowOneManyFetch(), type != Type.ATTRIBUTE);
+    detail.markQueryJoins(beanDescriptor, lazyLoadManyPath, isAllowOneManyFetch(), type != Type.ATTRIBUTE && type != Type.SUBQUERY);
```

Comment:
Checkme: Should we also check for other types here (e.g. `Type.ID_LIST`)?

@rbygrave There are several ways how Ebean detects if we have a subQuery:
See https://github.com/ebean-orm/ebean/blob/master/ebean-core/src/main/java/io/ebeaninternal/server/query/SqlTreeBuilder.java#L105
you probably know better what's right here